### PR TITLE
feat(description-list): added hor/vert responsive mods

### LINKED ISSUE #3630: (RFE) DescriptionList: missing option where the horizonal list can become vertical on small screens
In cockpit we would like to use isHorizontal for description lists (1 column), but we would like that they are responsive and convert into vertical for smaller screen sizes.
It would be useful to provide an option for that.

So instead of boolean it would be nice to have: isHorizontal: { "lg" } or something similar.

### PR DESCRIPTION
**Pull request by mattnolting**
closes #3630

### CODE REVIEW

File: src/patternfly/components/DescriptionList/examples/DescriptionList.md [hunk 1]
```diff
@@ -109,10 +114,11 @@ cssPrefix: pf-c-description-list
...
 | `.pf-c-description-list__text` | `<span>`, `<div>` | Initiates a description list component text element. Use a `<span>` when a child of `.pf-c-description-list__term`. **Required** |
-| `.pf-m-horizontal` | `.pf-c-description-list` | Modifies the description list component term and description pair to a horizontal layout. |
+| `.pf-m-horizontal{-on-[sm, md, lg, xl, 2xl]}` | `.pf-c-description-list` | Modifies the description list component term and description pair to a horizontal layout. |
+| `.pf-m-vertical{-on-[sm, md, lg, xl, 2xl]}` | `.pf-c-description-list` | Modifies the description list component term and description pair to a vertical layout. |
 | `.pf-m-auto-column-widths` | `.pf-c-description-list` | Modifies the description list to format automatically. |
 | `.pf-m-inline-grid` | `.pf-c-description-list` | Modifies the description list display to inline-grid. |
-| `.pf-m-{1,2,3}-col{-on-[md, lg, xl, 2xl]}` | `.pf-c-description-list` | Modifies the description list number of columns. |
+| `.pf-m-{1,2,3}-col{-on-[sm, md, lg, xl, 2xl]}` | `.pf-c-description-list` | Modifies the description list number of columns. |
```
Comment (mcoker) [inline src/patternfly/components/DescriptionList/examples/DescriptionList.md:121]: Do you know if there was a reason we didn't support this on the small breakpoint before?

**Review by mcoker — APPROVED** [review]
L👍TM!

File: src/patternfly/components/DescriptionList/examples/DescriptionList.md (hunk 1 above)
Comment (mattnolting) [reply src/patternfly/components/DescriptionList/examples/DescriptionList.md:121]: It was a decision that I made in effort to limit file size, but it's not a huge savings and probably is unexpected for devs.

**Comment by mattnolting** [comment]
@KKoukiou

**Pull request merged**